codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2FrameReader.java

```diff
@@ -635,8 +642,11 @@ final void close() {
          *            This is used for an optimization for when the first fragment is the full
          *            block. In that case, the buffer is used directly without copying.
          */
-        final void addFragment(ByteBuf fragment, ByteBufAllocator alloc, boolean endOfHeaders) {
+        final void addFragment(ByteBuf fragment, ByteBufAllocator alloc, boolean endOfHeaders) throws Http2Exception {
             if (headerBlock == null) {
+                if (fragment.readableBytes() > headersDecoder.configuration().maxHeaderSize()) {
+                    headerSizeExceeded();
```

Comment:
Just to be consistent with the other checks, maybe make a method `verifyHeaderSize(ByteBuf fragment)` that performs the check and then throws if the new fragment violates the threshold?